invoices/update.go

```diff
@@ -223,9 +235,6 @@ func updateLegacy(ctx *invoiceUpdateCtx,
 		return nil, ctx.failRes(ResultAmountTooLow), nil
 	}
 
-	// TODO(joostjager): Check invoice mpp required feature
```

Comment:
An alternative implementation here would be to carry out this TODO and still allow this path if we're processing an invoice that still had the _optional_ bit set.

Reply:
IMO this is the preferred option, to use the invoice as the source of truth. For invoices generated pre-0.12, a payment-secret-unaware node would ignore the feature bits and be able to complete a legacy payment. But if the receiver upgrades they will start getting an ambiguous payment error after making the attempt. IMO the invoice should work identically before and after the receiver's update. We also can't be sure how long people are setting their expiries for, so this very well could happen in practice.

Reply:
Going with this direction, resulted in a smaller diff overall as less of the tests needed to be modified.